src/dragonball/src/tracer.rs

```diff
@@ -0,0 +1,287 @@
+    /// Tracer name.
+    pub sid: String,
+    /// If enabled, the hpyervisor will create opentracing.io traces and spans.
+    /// See https://www.jaegertracing.io/docs/getting-started.
+    /// The full url to the Jaeger HTTP Thrift collector.
+    pub jaeger_endpoint: String,
+
+    /// The username to be used if basic auth is required for Jaeger.
+    pub jaeger_user: String,
+
+    /// The password to be used if basic auth is required for Jaeger.
+    pub jaeger_password: String,
+}
+
+impl Debug for TraceConfigInfo {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TraceConfigInfo")
+            .field("sid", &self.sid)
+            .field("jaeger_endpoint", &self.jaeger_endpoint)
+            .field("jaeger_user", &self.jaeger_user)
+            .field("jaeger_password", &self.jaeger_password)
+            .finish()
+    }
+}
+
+impl PartialEq for TraceConfigInfo {
+    fn eq(&self, other: &Self) -> bool {
+        if self.sid == other.sid
+            && self.jaeger_endpoint == other.jaeger_endpoint
+            && self.jaeger_user == other.jaeger_user
+            && self.jaeger_password == other.jaeger_password
+        {
+            return true;
+        }
+        false
+    }
+}
+
+impl Eq for TraceConfigInfo {}
+
+/// drabonball tracer.
+/// The dragonball tracer can run in two situations:
+///     1. dragonball runs as lib, such as running in kata.
+///     2. dragonball runs as app, such as running with `dbs-cli`.
+pub struct DragonballTracer {
+    /// Dragonball tracing subscriber.
+    subscriber: Arc<dyn Subscriber + Send + Sync>,
+    /// Subscriber guard. The lifetime is as same as tracer.
+    sub_guard: Option<DefaultGuard>,
+    /// The subscriber obtained from runtime.
+    from_runtime: bool,
+    /// Enable tracing
+    enable_tracing: bool,
+    /// Tracing's async runtime and lifetime is as same as tracer.
+    tokio_rt: Runtime,
+    /// Dragonball tracing root span.
+    root_span: Option<Arc<EnteredSpan>>,
+}
+
+impl Default for DragonballTracer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl DragonballTracer {
+    /// Create a dragonball tracer instance.
+    pub fn new() -> Self {
+        Self {
+            subscriber: Arc::new(NoSubscriber::default()),
+            sub_guard: None,
+            from_runtime: false,
+            enable_tracing: false,
+            tokio_rt: tokio::runtime::Builder::new_multi_thread()
+                .enable_all()
+                .build()
+                .expect("Tracer: tokio new thread err"),
+            root_span: None,
+        }
+    }
+
+    /// Return whether the tracing is enabled, enabled by [`setup_tracing`]
+    pub fn enabled(&self) -> bool {
+        self.enable_tracing
+    }
+
+    /// tracing feature for app.
+    pub fn setup_tracing(&mut self, config: TraceConfigInfo) -> Result<(), DragonballTraceError> {
+        if !self.enable_tracing {
```

Comment:
enable_tracing indicates a switcher, not the status, you can use tracing_on

Reply:
Fixed.